git_machete/git_operations.py

```diff
@@ -206,8 +208,9 @@ def __init__(self) -> None:
 
     @staticmethod
     def _run_git(git_cmd: str, *args: str, **kwargs: Any) -> int:
+        allow_non_zero = kwargs.pop("allow_non_zero", False)
         exit_code = utils.run_cmd("git", git_cmd, *args, **kwargs)
-        if not kwargs.get("allow_non_zero") and exit_code != 0:
```

Comment:
Also, I'd avoid modifying params to function **in place**. This might have unintended side effects, if the param object is later used by the caller. Make a copy, and remove the `"allow_non_zero"` key from that copy.